exercise/api/full_serializers.py

```diff
@@ -181,8 +182,12 @@ class Meta(AplusSerializerMeta):
             'exercise',
             'grading_data',
             'is_graded',
+            'feedback_response_seen'
         )
 
+    def get_feedback_response_seen(self, obj):
+        return not obj.notifications.filter(seen=False).exists() and obj.notifications.count() > 0
```

Comment:
Out of curiosity, is it necessary to here also check that the notification count is greater than zero? Does it not work if that is left out?

Reply:
Without this check, the submission is immediately set as feedback_response_seen = true, as there are no feedback responses (and no unseen responses).

Reply:
Ah right, that makes sense! Perhaps that could be shortly commented in the code (or there could be even a minor documentation string for the function). You could also add type hints at the same time.